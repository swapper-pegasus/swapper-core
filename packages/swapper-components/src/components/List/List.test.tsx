import React from 'react'
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer'
import { List, Props } from '.'

const props: Props = {
  headers: [
    { name: 'header 1', parts: 4 },
    { name: 'header 2', parts: 3 },
    { name: 'header 3', parts: 3 },
    { name: 'header 4', parts: 2 }
  ],
  rows: [
    ['Elemente 1,1', 'Elemente 1,2', 'Elemente 1,3', 'Elemente 1,4'],
    ['Elemente 2,1', 'Elemente 2,2', 'Elemente 2,3', <div><button>Button 1</button><button>Button 2</button></div>],
    ['Elemente 3,1', 'Text long for field parts, Text long for field parts', 'Elemente 3,3', 'Elemente 3,4'],
    ['Elemente 4,1', <p className='text-swpGray'>Custom element</p>, 'Elemente 4,3', 'Elemente 4,4']
  ]
}

describe('List', () => {
  it('render', () => {
    const tree = renderer
      .create(<List {...props}/>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
