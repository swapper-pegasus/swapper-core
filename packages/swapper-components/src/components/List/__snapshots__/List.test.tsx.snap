// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List render 1`] = `
<div
  className="w-full"
>
  <div
    className="flex"
  >
    <div
      className="w-4/12 border-b"
    >
      <p
        className="font-light text-swpGray-light p-2"
      >
        header 1
      </p>
    </div>
    <div
      className="w-3/12 border-b"
    >
      <p
        className="font-light text-swpGray-light p-2"
      >
        header 2
      </p>
    </div>
    <div
      className="w-3/12 border-b"
    >
      <p
        className="font-light text-swpGray-light p-2"
      >
        header 3
      </p>
    </div>
    <div
      className="w-2/12 border-b"
    >
      <p
        className="font-light text-swpGray-light p-2"
      >
        header 4
      </p>
    </div>
  </div>
  <div>
    <div
      className="flex w-full"
    >
      <div
        className="w-4/12 border-b p-2"
      >
        Elemente 1,1
      </div>
      <div
        className="w-3/12 border-b p-2"
      >
        Elemente 1,2
      </div>
      <div
        className="w-3/12 border-b p-2"
      >
        Elemente 1,3
      </div>
      <div
        className="w-2/12 border-b p-2"
      >
        Elemente 1,4
      </div>
    </div>
    <div
      className="flex w-full"
    >
      <div
        className="w-4/12 border-b p-2"
      >
        Elemente 2,1
      </div>
      <div
        className="w-3/12 border-b p-2"
      >
        Elemente 2,2
      </div>
      <div
        className="w-3/12 border-b p-2"
      >
        Elemente 2,3
      </div>
      <div
        className="w-2/12 border-b p-2"
      >
        <div>
          <button>
            Button 1
          </button>
          <button>
            Button 2
          </button>
        </div>
      </div>
    </div>
    <div
      className="flex w-full"
    >
      <div
        className="w-4/12 border-b p-2"
      >
        Elemente 3,1
      </div>
      <div
        className="w-3/12 border-b p-2"
      >
        Text long for field parts, Text long for field parts
      </div>
      <div
        className="w-3/12 border-b p-2"
      >
        Elemente 3,3
      </div>
      <div
        className="w-2/12 border-b p-2"
      >
        Elemente 3,4
      </div>
    </div>
    <div
      className="flex w-full"
    >
      <div
        className="w-4/12 border-b p-2"
      >
        Elemente 4,1
      </div>
      <div
        className="w-3/12 border-b p-2"
      >
        <p
          className="text-swpGray"
        >
          Custom element
        </p>
      </div>
      <div
        className="w-3/12 border-b p-2"
      >
        Elemente 4,3
      </div>
      <div
        className="w-2/12 border-b p-2"
      >
        Elemente 4,4
      </div>
    </div>
  </div>
</div>
`;
